File: Data Structures & Algorithms/binary-tree-vertical-order-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private static class Entry {
        TreeNode node;
        int col;

        Entry(TreeNode node, int col) {
            this.node = node;
            this.col = col;
        }
    }

    public List<List<Integer>> verticalOrder(TreeNode root) {
        if (root == null) return new ArrayList<>();

        Map<Integer, List<Integer>> cols = new TreeMap<>();
        Queue<Entry> q = new LinkedList<>();

        q.offer(new Entry(root, 0));

        while(!q.isEmpty()) {
            Entry e = q.poll();

            cols.computeIfAbsent(e.col, k -> new ArrayList<>()).add(e.node.val);

            if (e.node.left != null) {
                q.offer(new Entry(e.node.left, e.col - 1));
            }
            if (e.node.right != null) {
                q.offer(new Entry(e.node.right, e.col + 1));
            }
        }

        return new ArrayList<>(cols.values());
    }
} 

File: Data Structures & Algorithms/find-leaves-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def findLeaves(self, root: Optional[TreeNode]) -> List[List[int]]:
        self.res = []

        def dfs(node):
            if node is None:
                return -1

            leftHeight = dfs(node.left)
            rightHeight = dfs(node.right)

            currHeight = max(leftHeight, rightHeight) + 1

            if len(self.res) == currHeight:
                self.res.append([])

            self.res[currHeight].append(node.val)
            return currHeight

        dfs(root)
        return self.res